pilot/pkg/bootstrap/servicecontroller.go

```diff
@@ -92,7 +92,8 @@ func (s *Server) initKubeRegistry(args *PilotArgs) (err error) {
 		s.fetchCARoot,
 		s.environment,
 		s.environment.ClusterLocal(),
-		s.server)
+		s.server,
+		&s.environment.ClusterSet)
```

Comment:
We do pass s.environment already - methods with that many params are hard to read and fragile.

Reply:
I've moved this out of the environment, since it's really kube-specific. I agree that this argument list has gotten way too long, but I'd rather not add more to this PR, which is already long.

Reply:
This is no longer an issue.